lib/puppetserver/ca/config/puppet.rb

```diff
@@ -66,8 +66,9 @@ def load(cli_overrides = {})
           results ||= {}
           results[:main] ||= {}
           results[:master] ||= {}
+          results[:agent] ||= {}
 
-          overrides = results[:main].merge(results[:master])
+          overrides = results[:agent].merge(results[:main]).merge(results[:master])
```

Comment:
I'm not sure if I like this or not. This CA client in my mind isn't really associated with the puppet agent. Is there a specific reason you want to do this?

Reply:
Yeah, I think ideally it would read just the [main] section (like apply and a lot of puppet subcommands).

I think we'll break a bunch of folks changing the default from $certname to puppet unless we also read from the agent section. I have yet to find a test/dev set up that set server in [main] (though many set it in [agent]) which makes sense, and in all those cases it just worked before because the server in the agent section on the master was set to $certname.

Reply:
More concisely, I don't think this is correct, but I don't think it is any less correct than also merging in the master section, and I think it will meaningfully ease folks' upgrade.

Reply:
Got it. Fair enough.

Fun fact, most of Puppet's CLI tools actually will prefer the little known `user` section over `[main]`: https://puppet.com/docs/puppet/5.4/config_file_main.html#config-sections. Found this when I assumed they'd use `[agent]` and couldn't figure out why it wasn't working. Since I think almost no one uses it though, I don't think we have account for it here.

Reply:
I think there is, or was, a ca section too?

Reply:
Haven't heard of that one.